catroid/src/test/java/org/catrobat/catroid/test/content/actions/WebRequestActionTest.java

```diff
@@ -185,4 +190,50 @@ public void testCancelledCallAndResendRequest() {
 		verify(StageActivity.stageListener.webConnectionHolder, times(2)).addConnection(any());
 		verify(webConnection, times(2)).sendWebRequest();
 	}
+
+	@Test
+	public void testSuccessfulResponseWithInputVariable() {
+		UserVariable inputVariable = new UserVariable(TEST_INPUTVARIABLE, TEST_URL);
+		FormulaElement formulaElement = new FormulaElement(
+				FormulaElement.ElementType.USER_VARIABLE, inputVariable.getName(), null);
+		final String responseString = "Response";
+
+		Formula formula = Mockito.spy(new Formula(formulaElement));
+
+		testSprite.addUserVariable(userVariable);
+		testSprite.addUserVariable(inputVariable);
+
+		WebRequestAction action = (WebRequestAction) testSprite.getActionFactory().createWebRequestAction(
+				testSprite,
+				formula,
+				userVariable
+		);
+		action.setWebConnectionFactory(webConnectionFactory);
+
+		when(StageActivity.stageListener.webConnectionHolder.addConnection(any())).thenReturn(true);
+
+		try {
+			doAnswer(invocation -> {
+				Sprite sprite = invocation.getArgument(0);
+				return String.valueOf(sprite.getUserVariable(TEST_INPUTVARIABLE).getValue());
+			}).when(formula).interpretString(any(Sprite.class));
+		} catch (InterpretationException e) {
+			fail("Formula could not be interpreted");
+			return;
+		}
+
+		doAnswer(invocation -> {
+			String url = invocation.getArgument(0);
+			assertEquals(url, TEST_URL);
+			return webConnection;
+		}).when(webConnectionFactory).createWebConnection(anyString(), any());
+
+		doAnswer(invocation -> {
+			action.onRequestFinished(responseString);
+			return null;
+		}).when(webConnection).sendWebRequest();
+
+		assertTrue(action.act(0f));
+		assertEquals(responseString, userVariable.getValue());
```

Comment:
Split these asserts up in several TCs as far as possible. If a setup function is implemented, for each Testcase your setup will be loaded and then its easier to split up

Reply:
Done.